hazelcast-hibernate/hazelcast-hibernate4/src/main/java/com/hazelcast/hibernate/local/LocalRegionCache.java

```diff
@@ -195,18 +193,28 @@ protected Object createMessage(final Object key, final Object value, final Objec
     protected MessageListener<Object> createMessageListener() {
         return new MessageListener<Object>() {
             public void onMessage(final Message<Object> message) {
+                if (message.getPublishingMember().localMember()) {
+                    return;
+                }
                 final Invalidation invalidation = (Invalidation) message.getMessageObject();
-                if (versionComparator != null) {
-                    final Expirable value = cache.get(invalidation.getKey());
-                    if (value != null) {
-                        Object currentVersion = value.getVersion();
-                        Object newVersion = invalidation.getVersion();
-                        if (versionComparator.compare(newVersion, currentVersion) > 0) {
-                            cache.remove(invalidation.getKey(), value);
+                Object key = invalidation.getKey();
+                if (key != null) {
```

Comment:
> can you explain why you clear the entire cache on a `null` key?

I'm afraid this is how Hibernate works.  On a call to `unlockRegion`

```
 * There is another usage pattern that is used to invalidate entries
 * after performing "bulk" HQL/SQL operations:
 * {@link #lockRegion} -> {@link #removeAll} -> {@link #unlockRegion}
```

The cache provider is expected to clear all cached items in that region, because Hibernate doesn't know precisely which entities/collections may have been modified and we have to err on the side of caution.  And the `LocalRegionCache` has to apply this to all cluster nodes. 

The only place an event is published with a `null` key is in `LocalRegionCache.clear()`.  I suppose I could have created a new topic for these, but overloading the `null` key to mean "all keys" made sense to me, and cuts down the number of topics.